root/scripts/createRecipe.js

```diff
@@ -1,175 +1,209 @@
+    }
   }
-  const ingStep = document.getElementById(`ing${ingCount.toString()}`);
-  const amountStep = document.getElementById(`amount${ingCount.toString()}`);
-  const unitStep = document.getElementById(`units${ingCount.toString()}`);
-  unitStep.remove();
-  ingStep.remove();
-  amountStep.remove();
 }
 
-async function fillRecipePage(recipeId) {
-  const recipe = await readRecipe(recipeId);
-  const header = document.getElementById('header');
-  header.innerHTML = 'Edit Your Recipe!';
-  const name = document.getElementById('name');
-  name.value = recipe.title;
-  const imageLink = document.getElementById('image');
-  imageLink.value = recipe.image;
-  const summary = document.querySelector('.descrip');
-  summary.value = recipe.summary.replace(/<[^>]+>/g, '');
-  const servings = document.getElementById('serving');
-  servings.value = recipe.servings;
-  const time = document.getElementById('time');
-  time.value = recipe.readyInMinutes;
-  for (let j = 1; j < recipe.ingredients.length + 1; j += 1) {
-    addIng();
-    const ingredientName = document.getElementById(`ing${j.toString()}`);
-    const amount = document.getElementById(`amount${j.toString()}`);
-    const unit = document.getElementById(`units${j.toString()}`);
-    amount.value = recipe.ingredients[j - 1].amount;
-    ingredientName.value = recipe.ingredients[j - 1].name;
-    unit.value = recipe.ingredients[j - 1].unit;
-  }
+String.prototype.replaceAt = function (index, replacement) {
+  return this.substr(0, index) + replacement + this.substr(index + replacement.length);
+};
 
-  for (let k = 1; k <= recipe.steps.length; k += 1) {
-    if (k > 5) {
-      addStep();
+function getNum(str, multiple) {
+  let newStr = '';
+  if (str.length === 0) {
+    return '';
+  }
+  let numStr = '';
+  let i = 0;
+  while (true) {
+    if (str[i] === ' ') {
+      break;
     }
-    const stepVal = document.getElementsByClassName('step')[k - 1];
-    stepVal.value = recipe.steps[k - 1].step;
+    numStr += str[i];
+    i += 1;
+  }
+  let num = parseFloat(numStr);
+  num *= multiple;
+  const newNumStr = num.toString();
+  for (let i = 0; i < newNumStr.length; i++) {
+    newStr += newNumStr[i];
+  }
+  for (let i = numStr.length; i < str.length; i++) {
+    newStr += str[i];
+  }
+  // console.log(multiple);
+  console.log(newStr);
+  return newStr;
+}
+
+function scaleIngredients() {
+  // console.log(`this is the numArr: ${numArr[0].innerText}`);
+  const scale = document.getElementById('servings');
+  // console.log(scale);
+  const newRecipe =[];
+  const recipeIngredientsElement = document.getElementsByClassName('ingred');
+  for (let i = 0; i < recipeIngredientsElement.length; i++) {
+    
+    const newStr = getNum(recipeIngredientsElement[i].innerText.toString(), scale.value);
+    // console.log(recipeIngredientsElement[i].innerText);
+    // recipeIngredientsElement[i].innerText = newStr;
+    newRecipe [i].innerText = recipeIngredientsElement[i].innerText;
```

Comment:
Codacy has a fix for the issue: [Unexpected whitespace before property i.](https://app.codacy.com/gh/cse110-fa21-group34/rocketrecipes/pullRequest?prid=8487889)
```suggestion
    newRecipe[i].innerText = recipeIngredientsElement[i].innerText;
```